src/stan_math_backend/Transform_Mir.ml

```diff
@@ -764,9 +764,27 @@ let add_reads vars mkread stmts =
   let var_names = String.Map.of_alist_exn vars in
   let add_read_to_decl (Stmt.Fixed.{pattern; _} as stmt) =
     match pattern with
-    | Decl {decl_id; _} when Map.mem var_names decl_id ->
+    | Decl {decl_id; decl_adtype; decl_type; _} when Map.mem var_names decl_id
+      -> (
         let loc, out = Map.find_exn var_names decl_id in
-        stmt :: mkread (Stmt.Helpers.lvariable decl_id, loc, out)
+        let param_reader = mkread (Stmt.Helpers.lvariable decl_id, loc, out) in
+        match param_reader with
+        | [ Stmt.Fixed.
```

Comment:
This looks pretty difficult to maintain and should probably be folded into mkread somehow